File: oshi-front/src/apis/request/sale/AddSale.Request.Dto.ts
export default interface AddSaleRequestDto {
    oshiId: number; // @NotNull -> number 타입
    categoryId: number; // @NotNull -> number 타입
    goodsId: number; // @NotNull -> number 타입
    price: number; // BigDecimal을 number로 변환
    description: string; // @NotNull -> string 타입
    file?: File; // MultipartFile -> File, 선택 필드로 변환
  }
  

File: ohsi-back/src/main/java/com/oshi/ohsi_back/domain/sale/presentation/dto/request/AddSaleRequestDto.java
package com.oshi.ohsi_back.domain.sale.presentation.dto.request;

import jakarta.validation.constraints.NotNull;

import org.springframework.lang.Nullable;
import org.springframework.web.multipart.MultipartFile;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AddSaleRequestDto {

    @NotNull
    private int oshiId;
    @NotNull
    private int categoryId;
    @NotNull
    private int goodsId;
    @NotNull
    private BigDecimal price;
    @NotNull
    private String description;
    @Nullable
    MultipartFile file;

}
